.NET/Microsoft.Recognizers.Text.DateTime/Models/DateTimeModel.cs

```diff
@@ -61,7 +61,9 @@ public List<ModelResult> Parse(string query, System.DateTime refTime)
                 End = o.Start.Value + o.Length.Value - 1,
                 TypeName = o.Type,
                 Resolution = o.Value as SortedDictionary<string, object>,
-                Text = o.Text
+                Text = o.Text,
+                ParentText = (((o.Type.Split('.')).Last().Equals)(Constants.SYS_DATETIME_DATETIMEALT) ? 

```

Comment:
The right side here could be an auxiliary method. It would make the code easier to read.